### ISSUE TITLE
Import from garmin-forerunner-tools does not work on Ubuntu (Forerunner 305)

### ISSUE THREAD

**Issue by FreeCode4Freedom** [pytrainer/pytrainer#197]
Hello,

I have installed **pytrainer 2.0.0rc1** and **garmin-forerunner-tools 0.10repacked-11** on **Ubuntu 19.04**. After I enable "Garmin via garmintools" I get the folder "garmintools_full" in folder "plugins". When I close pytrainer and try to start then pytrainer does not start but trace:

```
running pytrainer from egg installation
data_path: /usr/share/pytrainer/
gettext_path: /usr/share/locale
site_path: /usr/lib/python2.7/site-packages
(pytr:13034): Gtk-WARNING **: 06:45:49.458: GtkFrame does not have a child property called padding
(pytr:13034): Gtk-WARNING **: 06:45:49.459: GtkFrame does not have a child property called expand
(pytr:13034): Gtk-WARNING **: 06:45:49.459: GtkFrame does not have a child property called fill
(pytr:13034): Gtk-WARNING **: 06:45:49.459: GtkFrame does not have a child property called position
Traceback (most recent call last):
  File "/usr/bin/pytr", line 98, in <module>
    main()
  File "/usr/bin/pytr", line 95, in main
    pytrainer = pyTrainer(None, data_path)
  File "/usr/lib/python2.7/dist-packages/pytrainer/main.py", line 108, in __init__
    self.loadPlugins()
  File "/usr/lib/python2.7/dist-packages/pytrainer/main.py", line 200, in loadPlugins
    warnings.warn('Attempted to enable an unsupported plugin %s' % plugin_name,
NameError: global name 'warnings' is not defined
```

Any ideas?

Thanks & best regards.

**Comment by heiderich** [comment]
Probably this is caused because the package `warnings` is not imported. This was done in 73dbfc3f55b528754410f278bd17825a8bfef2af. You may want to try a newer version of pytrainer.

**Comment by FreeCode4Freedom** [comment]
@heiderich  Thanks a lot for your answere! I spent several hours, also installed the latest version which solved the issue regarding the "global name" ... warnings. But I could not find any way to get my data from the Garmin Forerunner into the pytrainer.
Even when I enable the the garmin tools it was written something like "Import from GPS device is not implemented yet".
The Import from GPX files genereate from a GPS Logger succeded, but the GPX files exported from the Garmin Forerunner crashed (shut down) the pyTrainer application.
Searching the net I could not find anyone using an old Garmin Forerunner with an actual pyTrainer ... 
I used pyTrainer for more than 8 years - looks like these days are gone.

**Comment by heiderich** [comment]
I am currently also unable to import data directly from a Forerunner 305. See my issue https://github.com/pytrainer/pytrainer/issues/192.

>  Even when I enable the the garmin tools it was written something like "Import from GPS device is not implemented yet".

Which version did you use? Could you please show the complete error message?

**Comment by FreeCode4Freedom** [comment]
@heiderich Here some lines:

Import => Plugins => Garmin via garmintools is enabled

Import => Import from GPS Device: It's written "Import from GPS Device is not yet implemented". When I choose "Test 2" at "Select your GPS device" I see "Tools". At "Garmintools" I see "Version: /usr/bin/garmin_save_runs" and in the next line "GPS device **not** found"

When I try to "Import from File" and choose the GPX file exported from the Garmin Forerunner 305 I get the error: "File /home ..../MyTracks.gpx is of unknown or unsupported file type."

Since I do not know about any plans of pytrainer development I have exported existing tracks and moved to another solution:

https://wp-osm-plugin.hanblog.net/tutorial/garmin-forerunner-305-with-gpsbabel-and-mytourbook-in-linux-ubuntu-19-04/ 

Br, Mike

**Comment by heiderich** [comment]
@FreeCode4Freedom This is probably not the most appropriate place for such a discussion, but:

Were you able to convert data from PyTrainer to MyTourbook?

**Comment by FreeCode4Freedom** [comment]
@FreeCode4Freedom I still love pyTrainer since it's simple and focused on what you need. It's a pity there is no support on this topic (you posted your issue a couple of months ago ...) and so I had to change.  If this issues get fixed, I will rethink it ... 
In 2010 I published a post [DE] about pyTrainer and posted also the place (home/<User>/.pytrainer/) where all tracks ("historical data") are stored: 
https://mika.hanblog.net/pytrainer-garmin-forerunner-305-mit-linux-ubuntu-10-04/ 
I simple imported all GPX files of this folder in MyTourbook and saved them to the MyTourbook database. 
So the answere is yes. But Im still struggling to get a simple nice views with pace/alt/bpm since MyTourbook brings much more than I need.

**Comment by viiru-** [comment]
As mentioned, this issue was fixed in commit 73dbfc3. The other part of the problem is being tracked in #192.
